crates/oxc_linter/src/config/oxlintrc.rs

```diff
@@ -218,7 +218,7 @@ impl Oxlintrc {
     /// Merges two [Oxlintrc] files together
     /// [Self] takes priority over `other`
```

Comment:
Is this comment still accurate? If not, please update it.

Given that this is a bit confusing (hence why implementation was wrong before this PR), it'd also be helpful to expand this comment with an example of what merging looks like - where `self` and `other` both have the same property.